### PR TITLE
Convert delays to instructions when assembling

### PR DESCRIPTION
**Pull request by lcapelluto**
### Summary

Previously, we could ignore delays rather than assembling them. Now, with pulse gates, there are times that we want to schedule ONLY a delay, and not including the delay itself would remove the delay!

### Details and comments
Example:
```
qc = QuantumCircuit(1, 1)
qc.x(0)
qc.append(Gate('example', 1, []), [0])
qc.measure(0, 0)
qc.add_calibration('example', [0], pulse.Delay(1600, pulse.DriveChannel(0)))
```

### CODE REVIEW

File: qiskit/qobj/converters/pulse_instruction.py [hunk 1]
```diff
@@ -341,6 +341,24 @@ def convert_shift_phase(self, shift, instruction):
         }
         return self._qobj_model(**command_dict)
 
+    @bind_instruction(instructions.Delay)
+    def convert_shift_phase(self, shift, instruction):
+        """Return converted `ShiftPhase`.
+
+        Args:
+            shift(int): Offset time.
+            instruction (ShiftPhase): Shift phase instruction.
```
Comment (taalexander) [inline qiskit/qobj/converters/pulse_instruction.py:350]: The name for this method needs to be updated.

**Review by taalexander — CHANGES_REQUESTED** [review]
Could we also add a standalone (no pulse gates) test of delay?

File: qiskit/qobj/converters/pulse_instruction.py (hunk 1 above)
Comment (lcapelluto) [reply qiskit/qobj/converters/pulse_instruction.py:350]: 😨 😱  So bad, thank you

File: test/python/compiler/test_assembler.py [hunk 2]
```diff
@@ -548,6 +548,16 @@ def test_pulse_gates_common_cals(self):
...
         self.assertFalse(hasattr(qobj.experiments[1].config, 'calibrations'))
 
+    def test_pulse_gates_delay_only(self):
+        """Test that a single delay gate is translated to an instruction."""
+        circ = QuantumCircuit(2)
+        circ.append(Gate('test', 1, []), [0])
+        circ.add_calibration('test', [0], pulse.Delay(160, DriveChannel(0)))
+        qobj = assemble(circ, FakeOpenPulse2Q())
+        self.assertEqual(len(qobj.config.calibrations.gates[0].instructions), 1)
+        self.assertEqual(qobj.config.calibrations.gates[0].instructions[0].to_dict(),
```
Comment (zachschoenfeld33) [inline test/python/compiler/test_assembler.py:558]: Maybe you can test w/ a `shift` to verify that works? Ie put an instruction before and make sure the time gets shifted.

**Review by taalexander — DISMISSED** [review]

**Review by zachschoenfeld33 — DISMISSED** [review]
LGTM

**Review by taalexander — APPROVED** [review]

**Pull request merged**